lib/macho/macho_file.rb

```diff
@@ -233,10 +264,12 @@ def dylib_id=(new_id)
       raise ArgumentError.new("new ID must be a String") unless new_id.is_a?(String)
       return unless dylib?
 
-      dylib_cmd = command(:LC_ID_DYLIB).first
-      old_id = dylib_id
+      old_cmd = command(:LC_ID_DYLIB).first
+      new_cmd = LoadCommand.create(:LC_ID_DYLIB, new_id, old_cmd.timestamp,
+        old_cmd.current_version, old_cmd.compatibility_version)
 
-      set_name_in_dylib(dylib_cmd, old_id, new_id)
+      add_command(new_cmd)
+      delete_command(old_cmd)
```

Comment:
Maybe I'm understanding this incorrectly, but I don't think `add_command`/command modification generalize well into `insert_command`. In the `add_command` case we do a straight replacement of the null bytes at the end of the load commands, while in the modification case we do an insertion of serialized data at an offset and then pad/depad at the end of the load commands. In order to reconcile these two, we'd have to re-calculate our position in the file and thereby spoil the abstraction of the offset.

(There's a good chance I'm understanding this incorrectly - I'm pretty tired)